### PR TITLE
Semimetrizability of multi-pointed lines

### PR DESCRIPTION
**Pull request by pzjp**
Completing issue #1039. Added P102 (semimetrizable) for
- Segment with 2 ends [S65](https://topology.pi-base.org/spaces/S000065)
- Line with 2 origins [S83](https://topology.pi-base.org/spaces/S000083)
- Line with $\aleph_0$ origins [S84](https://topology.pi-base.org/spaces/S000084)
- Line with uncountably many origins [S85](https://topology.pi-base.org/spaces/S000085)
- **(false)** Everywhere doubled line [S86](https://topology.pi-base.org/spaces/S000086)

The construction is described in detail for S85 only, since semimetrizability is a hereditary property.

PS. A similar argument as for S86 can be applied for the [Sorgenfrey line](https://topology.pi-base.org/spaces/S000043).

### CODE REVIEW

File: spaces/S000086/properties/P000102.md
Comment (prabau) [inline spaces/S000086/properties/P000102.md:14]:
```suggestion
An uncountable subset of $\mathbb R$ has an accumulation point (cf. {{mathse:310113}}), hence there exists a sequence of distinct values $(a_n)_{n=1}^\infty\subset A_m$ such that $a_n\to a\in \mathbb R$,
and consequently $(a_n,0)\to (a,1)$ in $X$. But $d((a_n,0),(a,1))\ge 1/m$ for all $n$, which is a contradiction.
```

**Comment by prabau** [comment]
Nice proof for the "Everywhere doubled line".  I just added a minor suggestion.
It seems to me that space would be symmetrizable, right?

**Comment by yhx-12243** [comment]
> Nice proof for the "Everywhere doubled line". I just added a minor suggestion. It seems to me that space would be symmetrizable, right?

s86p104

**Comment by prabau** [comment]
Thanks @yhx-12243

**Comment by prabau** [comment]
@pzjp Before we finish this particular PR, would you mind reviewing #1140 as it's also related to semimetrizability, and based on your suggestion.
(The person who self-assigned as a reviewer for this is not available right now.)

**Comment by prabau** [comment]
This PR is now available in preview mode on branch `pr1156`.

**Comment by prabau** [comment]
This PR is using the fact that semimetrizable (P102) is a hereditary property.
We have started adding a "meta-properties" section to the property README files when it is used in proofs.

Could you add one for P102?  You can follow the pattern used in the README for P109.  I would say no justification needed as it's pretty obvious.

**Comment by prabau** [comment]
For the proof of semimetrizability for subspaces, we can use a a special pi-base syntax with single braces and a vertical bar:
`{S85|P102}` will be automatically expanded to "[space 85] is/is not [property 102]", with a link to corresponding page.

Example of use: https://topology.pi-base.org/spaces/S000029/properties/P000130

I'll do one suggestion for S84, and you can modify the other ones.

File: spaces/S000084/properties/P000102.md
Comment (prabau) [inline spaces/S000084/properties/P000102.md:7]:
```suggestion
{P102} is a hereditary property.
{S84} is a subspace of {S85}, and {S85|P102}.
```

**Comment by pzjp** [comment]
I did semimetrizability not symmetrizability since it is a bit easier to work with and they are equivalent for first countable.

Added another description of semimetrizability: the convergence description. It is used as the main definition in [Zbl 0116.14303](https://zbmath.org/0116.14303) as well.

**Comment by prabau** [comment]
> Added another description of semimetrizability: the convergence description. It is used as the main definition in [Zbl 0116.14303](https://zbmath.org/0116.14303) as well.

Regarding this "convergence description", I don't see where in that reference from Heath that description is given.  Where is it exactly?

**Comment by prabau** [comment]
Regarding the change to the previous second bullet of P102 (paragraph starting with "X is T1....").  There were three conditions that needed to be satisfied: (i) X is T1; (ii) And, for each ...; (iii) Moreover, ....
You combined (i) and (ii) into a single sentence.  So now it's not quite clear if that combined sentence is all there is to it, or if the Moreover part is also needed.  The previous version was slightly better, but I agree that it was not completely clear either.
I think it should be rephrased specifically with (i); (ii); (iii) to make everything clear.

And also the part "In particular, X is first countable" does not need to be there.  It's not part of the definition, it's a consequence of it.

But please don't change anything yet until we discuss your new second bullet first.

(Actually, when I saw you did this modification, I was hoping this would not derail this PR by requiring more discussion with more people later before this gets finalized.  That's exactly why we encourage people to keep the PRs small so that the non-controversial part can be released quickly.)

**Comment by prabau** [comment]
Regarding semimetrizability for S85 (line with uncountably many origins), I think the text seems a little confusing for people who rely on the definition from https://topology.pi-base.org/properties/P000102.

(the discussion in #959 is instructive to see how we got to that definition)

In more detail: "the sets $B_d(x,\varepsilon)$ ... are a base of topology on $X$."  Due to the (maybe ungrammatical?) lack of article, "base of topology" is not clear.  Is it meant as a base for any topology, namely the topology that it would generate as a subbase?  Or a base for the given topology of $X$?

"the semimetric topology": it is not clear what is meant by that.

But most of all, none of this talk about base for a topology should be relevant. We are given the topology of $X$ to start with.  That does not change.  And all we need to show is that for every point $x\in X$, the collection of balls $B(x,\varepsilon)$ form a **neighborhood base** at $x$.  That is easy to check and is exactly how the definition for P102 is phrased.

(Just to be clear: a collection of subsets of $X$ is called a neighborhood base at $x$ and it consists of neighborhoods of $x$ and every neighborhood of $x$ contains one member of the collection.  Completely standard stuff.)

----
Also, the last paragraph explaining that the same proof works for some other spaces is irrelevant and does not need to be here.  But we could instead link to the S85 proof from the other spaces, which would be simpler that the hereditary property argument.  Just saying something like: `Same proof as {S85|P102}.`

What do you think?

**Comment by pzjp** [comment]
> > Added another description of semimetrizability: the convergence description. It is used as the main definition in [Zbl 0116.14303](https://zbmath.org/0116.14303) as well.
> 
> Regarding this "convergence description", I don't see where in that reference from Heath that description is given. Where is it exactly?

At the very beginning Heath states (1), (2) and (3) 'axioms' for semimetric, with (3) being the equivalnece of convergence.

**Comment by pzjp** [comment]
Switched the proof for S85 to the neighbourhood base terminology. It is indeed less confusing. The hereditarity argument removed as well.

I indeed merged "1st" and "2nd" part of the last description of P102 but the "in particular {P28}" part was there already. I'm not sure if this condition is that important there at all. It looks like citing [metrization theorem](https://en.wikipedia.org/wiki/Nagata%E2%80%93Smirnov_metrization_theorem) for the definition of metrizability...

Exposed the points (i)-(iii) for now.

**Comment by yhx-12243** [comment]
> > Nice proof for the "Everywhere doubled line". I just added a minor suggestion. It seems to me that space would be symmetrizable, right?
> 
> s86p104

After #1161 (fix wrong T628), the automatically deduction of {S86|P104} become incorrect, so it'd be better gives a direct negation of P104.

**Comment by yhx-12243** [comment]
Oh S86 maybe Symmetrizable indeed (and become an another counterexample of original T628), just define $d((x, 0), (y, 0)) = \left| x - y \right|$, $d((x, 0), (x, 1)) = 1$, $d((x, 0), (y, 1)) = \left| x - y \right|$, $d((x, 1), (y, 1)) = 1$.

**Comment by pzjp** [comment]
@yhx-12243 your symmetric does not work. No point $(x,0)$ has a positive distance from the closed set $\mathbb R\times\\{1\\}$.
Originally I tried to disprove symmetrizability and reduced the problem to semimetrizability on my own. Then I skipped this part since _it was (supposedly) a general fact_. I need to recall the argument, though.

**Comment by prabau** [comment]
@pzjp You still have not provided a proper quote for your new "convergence description" bullet for P102. It may not even be true, I am not sure.  But in any case, it does not seem a helpful characterization to add.  The general guideline is we usually add an alternative characterization of a property when it's useful to prove some results here, or to prove some traits for spaces.  But we don't want to add all the possible equivalent formulations of something just because they exist.

Also, my criticism of the previous phrasing of the other bullet was not meant for us to fix right now. It was just to indicate that it needed to be improved (but not in this PR).

So in order to unblock this PR, I suggest to *not* modify anything further for P102.  I'll review the rest of the PR one more time.  And then I'll undo the changes we have here for P102. If desired, the merit of your new additional bullet can be debated separately, in further comments here even after the PR is merged.  Or in a separate issue or PR if you want.

**Review by prabau — CHANGES_REQUESTED** [review]
Please undo the changes you made to P102.  (I cannot do it myself, as it is in your repository.)

**Comment by yhx-12243** [comment]
> @yhx-12243 your symmetric does not work. No point ( x , 0 ) has a positive distance from the closed set R × { 1 } . Originally I tried to disprove symmetrizability and reduced the problem to semimetrizability on my own. Then I skipped this part since _it was (supposedly) a general fact_.
> 
> In fact the crucial property is: every sequence has a **finite** set of limits. Then, if x n → x but d ( x n , x ) > ϵ , the set c l ( { x n : n ≥ 1 } ) ∖ { x } has to be "semimetric closed" but is definitely not closed.

Fine. Since every sequence has at most two limits, the argument that “Sym + FU ⇒ Semi” actually works, is it fine to slightly modify the argument that gives a proof of {S86|¬P104}?

File: spaces/S000086/properties/P000104.md
Comment (prabau) [inline spaces/S000086/properties/P000104.md:16]:
```suggestion
If $d:X^2\to[0,+\infty)$ was a symmetric generating the topology, then for every $x\in\mathbb R$ there would exist $r_x>0$ such that $B_d((x,0),r_x)\subseteq \mathbb R\times\{0\}$.
Let $A_n:=\{x\in\mathbb R: r_x>1/n\}$.
Since $\mathbb R = \bigcup_{n=1}^\infty A_n$ is uncountable, so is $A_m$ for some $m\geq 1$.

An uncountable subset of $\mathbb R$ has an accumulation point (cf. {{mathse:310113}}), hence there exists a sequence of distinct values $(a_n)_{n=1}^\infty\subset A_m$ such that $a_n\to a\in \mathbb R$.
Let $S:=\overline{\{(a_n,0):n\in\mathbb N\}}=\{(a_n,0):n\in\mathbb N\}\cup\{(a,0),(a,1)\}$ and $S_0:=S\setminus\{(a,1)\}$.
Since $S$ is closed, every point $u\in X\setminus S$ satisfies $d(u,S_0)\ge d(u,S)>0$.
By definition of $A_m$, we also have $d((a_n,0),(a,1))\geq 1/m$, hence hence $d((a,1),S_0)>0$.
Since every point not in $S_0$ has a positive distance to $S_0$, $S_0$ is closed, which is a contradiction.
```

**Comment by prabau** [comment]
Nice proof for S86 (Everywhere doubled line) not being symmetrizable.

File: spaces/S000085/properties/P000102.md
Comment (prabau) [inline spaces/S000085/properties/P000102.md:7]:
```suggestion
Let $X=\mathbb{R}\backslash\{0\} \cup \{0_\alpha: \alpha\in S\}$ be a real line with multiple origins. Define a symmetric $d:X^2\to[0,+\infty)$ by
```
To match the definitions we use in pi-base (but please double check the previous discussions about that), such functions are called "symmetric", independently of any topology.  It is only after the fact, namely after showing it generates the topology with neighborhood bases that we can call it "semimetric" for the topology.

File: spaces/S000085/properties/P000102.md
Comment (prabau) [inline spaces/S000085/properties/P000102.md:15]: Since $x$ and $y$ were used in the previous paragraph to denote elements not equal to the origins, would it be clearer to use different letters here?  Up to you.

File: properties/P000102.md
Comment (prabau) [inline properties/P000102.md:28]: Need to restore this blank line.  Otherwise, see the bad effect in preview mode for branch `pr1156`.

File: spaces/S000083/properties/P000102.md
Comment (prabau) [inline spaces/S000083/properties/P000102.md:8]:
```suggestion
The proof remains the same.
```

File: spaces/S000084/properties/P000102.md
Comment (prabau) [inline spaces/S000084/properties/P000102.md:8]:
```suggestion
The proof remains the same.
```

File: spaces/S000085/properties/P000102.md
Comment (pzjp) [reply spaces/S000085/properties/P000102.md:15]: Now it's a and b in the definition.

**Comment by prabau** [comment]
@pzjp Looks good now.  I'll approve after this.  But wanted to ask one thing.

Did you ever explain to me where in Heath (https://doi.org/10.1090/S0002-9939-1961-0125562-9) there is mention of the characterization of semimetrizable in terms of convergence of arbitrary nets?

The only thing I can think of is condition (3) in the first paragraph of the paper:
"The topology of S is invariant with respect to the distance function d -- i.e., if p is a limit point of a subset M of S, then p is a distance limit point of M and conversely."

This looks like old language and not clear at all.  It does not seem to have anything to do with limits of arbitrary nets?  My guess would be that is means: $p\in\overline M$ if and only if $d(p,M)=0$, which is exactly one of the characterizations of semimetrizabily we already have.

But to make sure, I tried to check some other sources.  In particular, https://zbmath.org/0072.17802 (McAuley, 1956).  It uses exactly the same language and the same definition, without saying what limit point or distance limit point is exactly.  Not coincidentally, McAuley were both students of the same advisor, Burton Jones, who himself was a student of R.L. Moore.  So one would have to go back further to see the terminology commonly in use in the 1950's and earlier.  Maybe some earlier work of Moore?

Anyway, what is your understanding regarding this?

**Review by prabau — APPROVED** [review]

**Comment by pzjp** [comment]
I realised that while writting about the reference I misunderstood the Heath's condition (3). That's why I removed the convergence characterisation without further comment. Some old papers are not very precise when defining what is semimetrizability...

I guess convergence-definition should be equivalent in the same way as for metric spaces. But now I'm not sure whether lack of first countability may cause any problem...

**Pull request merged**
